piet-cairo/src/lib.rs

```diff
@@ -503,17 +511,24 @@ fn affine_to_matrix(affine: Affine) -> Matrix {
     }
 }
 
-fn compute_blurred_rect(rect: Rect, radius: f64) -> (ImageSurface, Point) {
+fn compute_blurred_rect(rect: Rect, radius: f64) -> Result<(ImageSurface, Point), cairo::Error> {
     let size = piet::util::size_for_blurred_rect(rect, radius);
-    // TODO: maybe not panic on error (but likely to happen only in extreme cases such as OOM)
-    let mut image =
-        ImageSurface::create(Format::A8, size.width as i32, size.height as i32).unwrap();
-    let stride = image.get_stride() as usize;
-    let mut data = image.get_data().unwrap();
-    let rect_exp = piet::util::compute_blurred_rect(rect, radius, stride, &mut *data);
-    std::mem::drop(data);
-    let origin = rect_exp.origin();
-    (image, origin)
+    match ImageSurface::create(Format::A8, size.width as i32, size.height as i32) {
+        Ok(mut image) => {
+            let stride = image.stride() as usize;
+            //TODO: This can still panic.
+            let mut data = image.data().unwrap();
```

Comment:
given that we've changed the signature to return an error, I think we can also address this now?

Reply:
It is sort of complicated. This returns a borrowerror, not a cairo::Error. I tried making the return type more general, but it cant really be anymore general. It would probably have to be a Box, but that wont work in `success`. If you know a way to do this, please let me know how.

Reply:
The only thing I can think of is that we store a `Box<dyn Error>` instead of a `cairo::Error`; we do end up converting to this type anyway when we make the `piet::Error::BackendError`?

Reply:
I just tried, but that doesnt work. I looked at the FFI code for `.data()` and it really seems impossible for it to return an error. An error is returned when either: the reference to `image` is dropped (it isnt since its still in scope), there is an error on `image` (there isnt since we havnt used it yet), the pointer to the image is null aka the surface isnt an imagesurface (it is an imagesurface), or the surface is finished (it isnt, we know because we dont finish it).

If someone else wants to remove this unwrap, im fine with it but its not worth it to me.

Reply:
Okay that sounds good, thanks for looking into it. It might be worth adding a little comment explaining that the unwrap shouldn't fail?